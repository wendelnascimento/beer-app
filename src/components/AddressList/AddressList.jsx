import React from 'react';
import PropTypes from 'prop-types';

import {
  List, ListItem, ItemTitle, ItemVicinity,
} from './styles';

const AddressList = ({ addresses }) => {
  if (addresses.length === 0) {
    return null;
  }

  return (
    <List>
      {addresses.map((item) => (
        <ListItem key={item.id}>
          <ItemTitle>{item.title}</ItemTitle>
          <ItemVicinity dangerouslySetInnerHTML={{ __html: item.vicinity }} />
        </ListItem>
      ))}
    </List>
  );
};

AddressList.propTypes = {
  addresses: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default AddressList;
