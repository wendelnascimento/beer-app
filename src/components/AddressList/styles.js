import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const List = styled.ul`
  list-style-type: none;
  padding: 16px 0;
  margin: 0;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  border-top: 1px solid #eee;
  box-shadow: rgba(0, 0, 0, 0.6) 0px 14px 23px -16px;
  background: #ffffff;
  max-height: 300px;
  overflow-y: scroll;
`;

export const ListItem = styled.li`
  padding: 8px 24px;
  width: 100%;

  &:not(:last-child) {
    border-bottom: 1px solid #fafafa;
    padding-bottom: 24px;
  }
`;

export const ItemTitle = styled.span`
  display: block;
  font-size: 1.2rem;
  color: #000000;
`;

export const ItemVicinity = styled.span`
  display: block;
  color: #cccccc;
  font-size: 1rem;
`;

export const CustomLink = styled(Link)`
  text-decoration: none;
`;
