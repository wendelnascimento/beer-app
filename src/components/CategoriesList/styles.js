import styled from 'styled-components';

export const List = styled.ul`
  list-style-type: none;
  padding: 0;
  margin: 0 0 16px;
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-gap: 16px;

  @media screen and (max-width: 900px) {
    grid-template-columns: repeat(5, 1fr);
  }

  @media screen and (max-width: 600px) {
    grid-template-columns: repeat(4, 1fr);
  }

  @media screen and (max-width: 460px) {
    grid-template-columns: repeat(3, 1fr);
  }

  @media screen and (max-width: 350px) {
    grid-template-columns: repeat(2, 1fr);
  }
`;

export const ListItem = styled.li`
  border: 2px solid #000;
  border-radius: 5px;
  overflow: hidden;
  transition: border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);

  &:hover {
    border-color: #f28e1c;
  }
`;

export const ItemButton = styled.button`
  width: 100%;
  padding: 8px 16px;
  border: none;
  font-weight: 700;
  text-align: center;
  background: #ffffff;

  &:hover {
    cursor: pointer;
  }
`;
