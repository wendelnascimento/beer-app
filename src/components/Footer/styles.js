import styled from 'styled-components';

export const Wrapper = styled.footer`
  padding: 16px;
  color: #cccccc;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  border-top: 1px solid #fafafa;
`;
