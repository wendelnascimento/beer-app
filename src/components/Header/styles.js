import styled from 'styled-components';

export const Wrapper = styled.header`
  max-width: 100vw;
  background: #fff;
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 1px solid #fafafa;
`;

export const Title = styled.h1`
  font-weight: 300;
  margin: 0;
`;
