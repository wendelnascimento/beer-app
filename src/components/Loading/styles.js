import styled, { keyframes } from 'styled-components';

export const bounceAndRotate = keyframes`
  0% {
    transform: scale3d(.8, .8, 1);
  }
  25% {
    transform: scale3d(1.5, 1.5, 1);
  }
  50% {
    transform: scale3d(.8, .8, 1);
  }
  75% {
    transform: rotate3d(0, 0, 1, 180deg);
  }
  100% {
    transform: scale3d(1, 1, 1), rotate3d(0, 0, 1, 360deg);
  }
`;

export const Wrapper = styled.div`
  position: fixed;
  z-index: 10;
  width: 100vw;
  height: 100vh;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(255, 255, 255, 0.6);
`;

export const Beer = styled.span`
  font-size: 7rem;
  transform: scale3d(1, 1, 1), rotate3d(0, 0, 1, 0deg);
  animation: ${bounceAndRotate} 2s infinite linear;
`;
