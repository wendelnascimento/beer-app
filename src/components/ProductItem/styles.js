import styled from 'styled-components';

export const Card = styled.li`
  width: 100%;
  max-width: 300px;
  background: #ffffff;
  border: 2px solid #fff;
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: rgba(0, 0, 0, 0.6) 0px 2px 15px -5px;
  transform: translate3d(0, 0, 0);
  transition: transform 0.3s cubic-bezier(0.19, 1, 0.22, 1),
    border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
  overflow: hidden;

  &:hover {
    transform: translate3d(0, -5px, 0);
    border-color: #f28e1c;
  }
`;

export const ProductImage = styled.img`
  width: 100%;
  object-fit: scale-down;
  background: #ffffff;
  border-radius: 5px;
  margin-bottom: 8px;
`;

export const ImagePlaceholder = styled.span`
  font-size: 5rem;
  text-align: center;
  margin-top: auto;
  margin-bottom: auto;
`;

export const ProductTitle = styled.span`
  color: #666;
  font-size: 0.8rem;
  margin-top: auto;
  padding: 0 8px;
`;

export const ProductPrice = styled.span`
  color: #f28e1c;
  font-size: 1.2rem;
  font-weight: 700;
  margin-bottom: 16px;
  padding: 0 8px;
`;

export const QuantityRow = styled.div`
  align-self: stretch;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const AddButton = styled.button`
  width: 30px;
  height: 30px;
  background: #f28e1c;
  color: #ffffff;
  border: none;
  font-size: 1.3rem;
  font-weight: 300;

  &:hover {
    cursor: pointer;
  }
`;

export const RemoveButton = styled(AddButton)`
  background: #b03c3c;
`;
