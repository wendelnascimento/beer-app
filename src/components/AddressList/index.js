import AddressList from './AddressList';

export default AddressList;
