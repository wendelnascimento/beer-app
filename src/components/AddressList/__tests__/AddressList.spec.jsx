import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter, Link } from 'react-router-dom';

import AddressList from '../AddressList';
import { ListItem } from '../styles';

describe('<AddressList />', () => {
  const addresses = [
    {
      id: 1,
      position: [123456, 456123],
      title: 'Test Address 1',
      vicinity: '<b>Test city 1</b>',
    },
    {
      id: 2,
      position: [123456, 456123],
      title: 'Test Address 2',
      vicinity: '<b>Test city 2</b>',
    },
    {
      id: 3,
      position: [123456, 456123],
      title: 'Test Address 3',
      vicinity: '<b>Test city 3</b>',
    },
    {
      id: 4,
      title: 'Test Address 4',
    },
  ];

  it('does render properly', () => {
    shallow(<AddressList addresses={[]} />);
  });

  it('matches stored snapshot with empty list', () => {
    const wrapper = mount(<AddressList addresses={[]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('matches stored snapshot with a populated list', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
        <AddressList addresses={addresses} />
      </MemoryRouter>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders only items with "location" property', () => {
    const wrapper = mount(
      <MemoryRouter>
        <AddressList addresses={addresses} />
      </MemoryRouter>,
    );

    expect(wrapper.find(ListItem)).toHaveLength(3);
  });

  it('renders null if there is no addresses', () => {
    const wrapper = mount(<AddressList addresses={[]} />);

    expect(wrapper.isEmptyRender()).toEqual(true);
  });

  it('contains links to list page', () => {
    const wrapper = mount(
      <MemoryRouter>
        <AddressList addresses={addresses} />
      </MemoryRouter>,
    );

    expect(
      wrapper
        .find(Link)
        .first()
        .prop('to'),
    ).toBe(
      `/list?lat=${addresses[0].position[0]}&long=${addresses[0].position[1]}&address=${addresses[0].title}`,
    );
  });
});
