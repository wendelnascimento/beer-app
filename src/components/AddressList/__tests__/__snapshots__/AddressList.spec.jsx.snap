// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressList /> matches stored snapshot with a populated list 1`] = `
.c0 {
  list-style-type: none;
  padding: 16px 0;
  margin: 0;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  border-top: 1px solid #eee;
  box-shadow: rgba(0,0,0,0.6) 0px 14px 23px -16px;
  background: #ffffff;
  max-height: 300px;
  overflow-y: scroll;
}

.c1 {
  padding: 8px 24px;
  width: 100%;
}

.c1:not(:last-child) {
  border-bottom: 1px solid #fafafa;
  padding-bottom: 24px;
}

.c3 {
  display: block;
  font-size: 1.2rem;
  color: #000000;
}

.c4 {
  display: block;
  color: #cccccc;
  font-size: 1rem;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <AddressList
      addresses={
        Array [
          Object {
            "id": 1,
            "position": Array [
              123456,
              456123,
            ],
            "title": "Test Address 1",
            "vicinity": "<b>Test city 1</b>",
          },
          Object {
            "id": 2,
            "position": Array [
              123456,
              456123,
            ],
            "title": "Test Address 2",
            "vicinity": "<b>Test city 2</b>",
          },
          Object {
            "id": 3,
            "position": Array [
              123456,
              456123,
            ],
            "title": "Test Address 3",
            "vicinity": "<b>Test city 3</b>",
          },
          Object {
            "id": 4,
            "title": "Test Address 4",
          },
        ]
      }
    >
      <styled.ul>
        <ul
          className="c0"
        >
          <styled.li
            key="1"
          >
            <li
              className="c1"
            >
              <Styled(Link)
                to="/list?lat=123456&long=456123&address=Test Address 1"
              >
                <Link
                  className="c2"
                  to="/list?lat=123456&long=456123&address=Test Address 1"
                >
                  <LinkAnchor
                    className="c2"
                    href="/list?lat=123456&long=456123&address=Test Address 1"
                    navigate={[Function]}
                  >
                    <a
                      className="c2"
                      href="/list?lat=123456&long=456123&address=Test Address 1"
                      onClick={[Function]}
                    >
                      <styled.span>
                        <span
                          className="c3"
                        >
                          Test Address 1
                        </span>
                      </styled.span>
                      <styled.span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<b>Test city 1</b>",
                          }
                        }
                      >
                        <span
                          className="c4"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "<b>Test city 1</b>",
                            }
                          }
                        />
                      </styled.span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li
            key="2"
          >
            <li
              className="c1"
            >
              <Styled(Link)
                to="/list?lat=123456&long=456123&address=Test Address 2"
              >
                <Link
                  className="c2"
                  to="/list?lat=123456&long=456123&address=Test Address 2"
                >
                  <LinkAnchor
                    className="c2"
                    href="/list?lat=123456&long=456123&address=Test Address 2"
                    navigate={[Function]}
                  >
                    <a
                      className="c2"
                      href="/list?lat=123456&long=456123&address=Test Address 2"
                      onClick={[Function]}
                    >
                      <styled.span>
                        <span
                          className="c3"
                        >
                          Test Address 2
                        </span>
                      </styled.span>
                      <styled.span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<b>Test city 2</b>",
                          }
                        }
                      >
                        <span
                          className="c4"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "<b>Test city 2</b>",
                            }
                          }
                        />
                      </styled.span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li
            key="3"
          >
            <li
              className="c1"
            >
              <Styled(Link)
                to="/list?lat=123456&long=456123&address=Test Address 3"
              >
                <Link
                  className="c2"
                  to="/list?lat=123456&long=456123&address=Test Address 3"
                >
                  <LinkAnchor
                    className="c2"
                    href="/list?lat=123456&long=456123&address=Test Address 3"
                    navigate={[Function]}
                  >
                    <a
                      className="c2"
                      href="/list?lat=123456&long=456123&address=Test Address 3"
                      onClick={[Function]}
                    >
                      <styled.span>
                        <span
                          className="c3"
                        >
                          Test Address 3
                        </span>
                      </styled.span>
                      <styled.span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<b>Test city 3</b>",
                          }
                        }
                      >
                        <span
                          className="c4"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "<b>Test city 3</b>",
                            }
                          }
                        />
                      </styled.span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
        </ul>
      </styled.ul>
    </AddressList>
  </Router>
</MemoryRouter>
`;

exports[`<AddressList /> matches stored snapshot with empty list 1`] = `
<AddressList
  addresses={Array []}
/>
`;
